index.js

```diff
@@ -2613,6 +2613,42 @@
   }
   S.ifElse = def('ifElse', {}, [Pred(a), Fn(a, b), Fn(a, b), a, b], ifElse);
 
+  //# when :: (a -> Boolean) -> (a -> b) -> a -> b
```

Comment:
I'm not sure on which version would be the most useful, but we should choose one which is "correct"